Deal with `(void)` function signature warnings

```c
* checking whether package ‘slider’ can be installed ... [5s/5s] WARNING
Found the following significant warnings:
  /home/Hornik/lib/R/Library/4.3/x86_64-linux-gnu/vctrs/include/vctrs.h:12:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  init.c:69:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  init.c:72:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  slider-vctrs-private.c:12:6: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  /home/Hornik/lib/R/Library/4.3/x86_64-linux-gnu/vctrs/include/vctrs.c:7:6: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  slider-vctrs-public.c:3:6: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  summary-core.h:9:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  summary-core.h:10:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
```

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,5 +1,7 @@
 # slider (development version)
 
+* Fixed `-Wstrict-prototypes` warnings as requested by CRAN (#173).
+
 * Bumped minimal version of R to >=3.4.0 to align with tidyverse standards.
 
 * Fixed redirecting URLs.

--- a/src/init.c
+++ b/src/init.c
@@ -66,10 +66,10 @@ void R_init_slider(DllInfo *dll)
 }
 
 // slider-vctrs-private.c
-void slider_initialize_vctrs_private();
+void slider_initialize_vctrs_private(void);
 
 // slider-vctrs-public.c
-void slider_initialize_vctrs_public();
+void slider_initialize_vctrs_public(void);
 
 // utils.c
 void slider_initialize_utils(SEXP);

--- a/src/slider-vctrs-private.c
+++ b/src/slider-vctrs-private.c
@@ -9,7 +9,7 @@ SEXP (*vec_set_names)(SEXP, SEXP) = NULL;
 SEXP (*compact_seq)(R_len_t, R_len_t, bool) = NULL;
 SEXP (*init_compact_seq)(int*, R_len_t, R_len_t, bool) = NULL;
 
-void slider_initialize_vctrs_private() {
+void slider_initialize_vctrs_private(void) {
   // Experimental non-public vctrs functions
   vec_cast = (SEXP (*)(SEXP, SEXP)) R_GetCCallable("vctrs", "exp_vec_cast");
   vec_chop = (SEXP (*)(SEXP, SEXP)) R_GetCCallable("vctrs", "exp_vec_chop");

--- a/src/slider-vctrs-public.c
+++ b/src/slider-vctrs-public.c
@@ -1,5 +1,5 @@
 #include <vctrs.c>
 
-void slider_initialize_vctrs_public() {
+void slider_initialize_vctrs_public(void) {
   vctrs_init_api();
 }

--- a/src/summary-core-align.cpp
+++ b/src/summary-core-align.cpp
@@ -10,11 +10,11 @@ extern "C" {
  * that.
  */
 
-size_t align_of_long_double() {
+size_t align_of_long_double(void) {
   return alignof(long double);
 }
 
-size_t align_of_mean_state_t() {
+size_t align_of_mean_state_t(void) {
   return alignof(struct mean_state_t);
 }
 

--- a/src/summary-core-align.h
+++ b/src/summary-core-align.h
@@ -7,8 +7,8 @@ extern "C" {
 
 #include "summary-core-types.h"
 
-size_t align_of_long_double();
-size_t align_of_mean_state_t();
+size_t align_of_long_double(void);
+size_t align_of_mean_state_t(void);
 
 } // extern "C"
 

--- a/src/summary-core.h
+++ b/src/summary-core.h
@@ -6,8 +6,8 @@
 #include "align.h"
 
 // From `summary-core-align.hpp`
-size_t align_of_long_double();
-size_t align_of_mean_state_t();
+size_t align_of_long_double(void);
+size_t align_of_mean_state_t(void);
 
 // -----------------------------------------------------------------------------
 // Sum

